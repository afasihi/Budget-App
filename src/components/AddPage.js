import React from 'react';
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses'

// export class AddPage extends React.Component {
// 	onSubmit = (expense) => {
// 		this.props.onSubmit(expense)
// 		this.props.history.push('/')
// 	}
// 	render() {
// 		return (
// 			<div>
// 				<h1>Add Expense</h1>
// 				<ExpenseForm
// 					onSubmit={this.onSubmit}
// 				/>
// 			</div>
// 		)
// 	}
// }

// const mapDispatchToProps = (dispatch) => ({
// 	onSubmit: (expense) => dispatch(addExpense(expense))
// })

// export default connect(undefined, mapDispatchToProps)(AddPage);

export class AddPage extends React.Component {
  onSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddPage);