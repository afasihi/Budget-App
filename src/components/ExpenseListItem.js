import React from 'react';
import { NavLink } from 'react-router-dom'
import moment from 'moment';
import numeral from 'numeral'

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
	<div>
		<NavLink to={`/edit/${id}`}>
			<h3>{description}</h3>
		</NavLink>
		<p>
			{numeral(amount / 100).format('$0,0.00')} 
			- 
			{moment(createdAt).format('MMMM Do, YYYY')}</p>

	</div>
)

export default ExpenseListItem