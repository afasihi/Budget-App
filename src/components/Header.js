import React from 'react';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth'
import { connect } from 'react-redux'

export const Header = ({startLogout}) => (
	<header>
		<h1>Expensify</h1>
		<NavLink to="/" activeClassName="is-active" exact={true}>Home page</NavLink>
		<NavLink to="/create" activeClassName="is-active" >Create page</NavLink>
		<button onClick={startLogout}>Logout</button>
	</header>
)

const mapDispatchToProps = (dispatch) => ({
	startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)