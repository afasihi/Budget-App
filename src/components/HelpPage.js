import React from 'react';

const HelpPage = () => (
	<div>
		tHis a test from help Page.
	</div>
)

export default HelpPage;