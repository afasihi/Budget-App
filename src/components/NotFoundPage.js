import React from 'react';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';

const NotFoundPage = () => (
	<div>
		404! - <Link to="/"> Home Page</Link>
	</div>
)

export default NotFoundPage;