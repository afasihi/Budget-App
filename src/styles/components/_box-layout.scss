.box-layout {
  background: url('/images/bg.jpg');
  background-size: cover;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
  width: 100vw;
}

.box-layout__box {
  background: fade-out($color: #fff, $amount: 0.20);
  border-radius: 3px;
  text-align: center;
  width: 25rem;
  padding: $l-size $m-size;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}