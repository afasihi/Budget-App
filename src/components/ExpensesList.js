import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
	<div className="content-container">
		<div className="list-header">
			<div className="show-for-mobile">Expenses</div>
			<div className="show-for-desktop">expense</div>
			<div className="show-for-desktop">Amount</div>
		</div>
		<div className="list-body">
			{
				props.expenses.length === 0 ? (
					<div className="list-item list-item--message">
						<span> no Expenses</span>
					</div>
				) : (
						props.expenses.map((expense) => {
							return <ExpenseListItem key={expense.id} {...expense} />;
						})
					)
			}
		</div>
	</div>
)

const mapStateToProps = (state) => {
	return {
		expenses: selectExpenses(state.expenses, state.filters)
	}
}

export default connect(mapStateToProps)(ExpenseList)