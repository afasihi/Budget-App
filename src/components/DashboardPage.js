import React from 'react';
import ExpensesList from './ExpensesList';
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummery from './ExpensesSummery'

const DashboardPage = () => (
	<div>
		<ExpensesSummery />
		<ExpenseListFilters />
		<ExpensesList />
	</div>
)

export default DashboardPage;