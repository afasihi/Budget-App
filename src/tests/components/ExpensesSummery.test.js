import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummery } from '../../components/ExpensesSummery'

test('should render ExpensesSummery correctly with 1 expense', () => {
	const wrapper = shallow(<ExpensesSummery expensesCount={1} expensesTotal={235}/>)
	expect(wrapper).toMatchSnapshot()
})
test('should render ExpensesSummery correctly with multiple expenses', () => {
	const wrapper = shallow(<ExpensesSummery expensesCount={54} expensesTotal={232235}/>)
	expect(wrapper).toMatchSnapshot()
})