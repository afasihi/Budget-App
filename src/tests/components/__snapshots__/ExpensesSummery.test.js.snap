// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummery correctly with 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h2>
</div>
`;

exports[`should render ExpensesSummery correctly with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    54
     
    expenses
     totalling 
    $2,322.35
  </h2>
</div>
`;
